include/nighthawk/common/request_source_plugin.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include "envoy/api/api.h"
+#include "envoy/common/pure.h"
+#include "envoy/config/typed_config.h"
+
+#include "nighthawk/common/request_source.h"
+
+namespace Nighthawk {
+
+
+
+/**
+ * A factory that must be implemented for each RequestSourcePlugin. It instantiates the specific
+ * RequestSourcePlugin class after unpacking the plugin-specific config proto.
+ */
+class RequestSourcePluginConfigFactory : public Envoy::Config::TypedFactory {
+public:
+  ~RequestSourcePluginConfigFactory() override = default;
+  std::string category() const override { return "nighthawk.request_source_plugin"; }
+  /**
+   * Instantiates the specific RequestSourcePlugin class. Casts |message| to Any, unpacks it to the
+   * plugin-specific proto, and passes the strongly typed proto to the plugin constructor.
+   *
+   * @param message Any typed_config proto taken from the TypedExtensionConfig.
+   *
+   * @param api Api parameter that contains timesystem, filesystem, and threadfactory.
+   *
+   * @return RequestSourcePtr Pointer to the new plugin instance.
+   *
+   * @throw Envoy::EnvoyException If the Any proto cannot be unpacked as the type expected by the
+   * plugin.
+   */
+  virtual RequestSourcePtr createRequestSourcePlugin(const Envoy::Protobuf::Message& message,
```

Comment:
It just might, but are we sure `api` always suffices as an input for every implementation to construct it?